### PR TITLE
Split EE and CE acceptance workflows

### PR DESCRIPTION
**Pull request by DonNicoJs**
Split CE and EE workflows, see conversation in staff discord for reasoning

### CODE REVIEW

File: .github/workflows/pr-acceptance-ee.yml [hunk 1]
```diff
@@ -0,0 +1,90 @@
+# This workflow runs acceptance tests on pull requests (both CE and EE). It needs to be run in the
+# target project instead of the fork in order to use secrets. This is why the actions/checkout
+# action regularly has to specify the pull request sha.
+#
+# SECURITY ADVISORY
+# Be careful while making changes to this file.
+# See: https://securitylab.github.com/research/github-actions-preventing-pwn-requests/
+
+name: pr-acceptance
```
Comment (DonNicoJs) [inline .github/workflows/pr-acceptance-ee.yml:9]:
```suggestion
name: pr-acceptance-ee
```

File: .github/workflows/pr-acceptance.yml [hunk 2]
```diff
@@ -8,6 +8,64 @@
 
 name: pr-acceptance
```
Comment (DonNicoJs) [inline .github/workflows/pr-acceptance.yml:9]:
```suggestion
name: pr-acceptance-ce
```

File: .github/workflows/pr-acceptance-ee.yml [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+        run: echo "::set-output name=defined::true"
+
+  acceptance-ee:
+    # Only run EE tests if the LICENSE_ENCRYPTION_PASSWORD secret exists, so that the workflow
+    # doesn't fail when code is pushed to a fork.
+    if: ${{ needs.license-encryption-password.outputs.defined }}
+    timeout-minutes: 60
+    runs-on: ubuntu-latest
+    needs: [go-version, license-encryption-password]
+    if: contains(github.event.pull_request.labels.*.name, 'safe to test')
```
Comment (DonNicoJs) [inline .github/workflows/pr-acceptance-ee.yml:62]:
```suggestion
```

File: .github/workflows/pr-acceptance-ee.yml [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+      - id: defined
+        env:
+          LICENSE_ENCRYPTION_PASSWORD: ${{ secrets.LICENSE_ENCRYPTION_PASSWORD }}
+        if: ${{ env.LICENSE_ENCRYPTION_PASSWORD != '' }}
+        run: echo "::set-output name=defined::true"
+
+  acceptance-ee:
+    # Only run EE tests if the LICENSE_ENCRYPTION_PASSWORD secret exists, so that the workflow
+    # doesn't fail when code is pushed to a fork.
+    if: ${{ needs.license-encryption-password.outputs.defined }}
```
Comment (DonNicoJs) [inline .github/workflows/pr-acceptance-ee.yml:58]:
```suggestion
    if: ${{ needs.license-encryption-password.outputs.defined }} && contains(github.event.pull_request.labels.*.name, 'safe to test')
```

File: .github/workflows/pr-acceptance-ee.yml [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+      - id: defined
+        env:
+          LICENSE_ENCRYPTION_PASSWORD: ${{ secrets.LICENSE_ENCRYPTION_PASSWORD }}
+        if: ${{ env.LICENSE_ENCRYPTION_PASSWORD != '' }}
+        run: echo "::set-output name=defined::true"
+
+  acceptance-ee:
+    # Only run EE tests if the LICENSE_ENCRYPTION_PASSWORD secret exists, so that the workflow
+    # doesn't fail when code is pushed to a fork.
+    if: ${{ needs.license-encryption-password.outputs.defined }} && contains(github.event.pull_request.labels.*.name, 'safe to test')
```
Comment (DonNicoJs) [inline .github/workflows/pr-acceptance-ee.yml:58]:
```suggestion
    if: needs.license-encryption-password.outputs.defined  && contains(github.event.pull_request.labels.*.name, 'safe to test')
```

**Comment by DonNicoJs** [comment]
@armsnyder when you have a moment can you review this? We are splitting the workflows, removing privileges from the CE one and executing the EE one only after we have a specific label on the PR

**Review by timofurrer — APPROVED** [review]

File: .github/workflows/pr-acceptance-ce.yml [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+jobs:
+  go-version:
+    runs-on: ubuntu-latest
+    outputs:
+      go-version: ${{ steps.go-version.outputs.go-version }}
+    steps:
+      # Check out the pull request code (as opposed to the target project).
+      - uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
```
Comment (armsnyder) [inline .github/workflows/pr-acceptance-ce.yml:28]: We can use the default settings here now.

```suggestion
      - uses: actions/checkout@v2
```

File: .github/workflows/pr-acceptance-ce.yml [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+    runs-on: ubuntu-latest
+    needs: [go-version]
+    steps:
+      - uses: actions/setup-go@v2
+        with:
+          go-version: ${{ needs.go-version.outputs.go-version }}
+      # Check out the pull request code (as opposed to the target project).
+      - uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
```
Comment (armsnyder) [inline .github/workflows/pr-acceptance-ce.yml:44]: Same as above.
```suggestion
      - uses: actions/checkout@v2
```

File: .github/workflows/pr-acceptance-ce.yml [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+      # Cache the Go modules.
+      - uses: actions/cache@v2
+        with:
+          path: ~/go/pkg/mod
+          key: ${{ github.job }}-${{ runner.os }}-go${{ env.GO_VERSION }}-${{ hashFiles('**/go.sum', 'GNUMakefile') }}
+      # CAUTION: EXECUTING UNTRUSTED CODE.
```
Comment (armsnyder) [inline .github/workflows/pr-acceptance-ce.yml:50]: Warning is a bit out of place now.
```suggestion
```

**Review by armsnyder — DISMISSED** [review]
Some minor refactor-y suggestions, but functionally this already looks correct, so approved!

**Review by armsnyder — APPROVED** [review]

**Pull request merged**